internal/terraform/evaluate.go

```diff
@@ -862,33 +873,11 @@ func (d *evaluationStateData) GetResource(addr addrs.Resource, rng tfdiags.Sourc
 		ret = val
 	}
 
-	// since the plan was not yet created during validate, the values we
-	// collected here may not correspond with configuration, so they must be
-	// unknown.
-	if d.Operation == walkValidate {
```

Comment:
This section was redundant with the default case before we checked the instance state, since there is no instance state during validation. Verified that this condition was no longer reached during tests.